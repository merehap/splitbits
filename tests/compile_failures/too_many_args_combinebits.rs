use splitbits::*;

fn main() {
    let x: u16 = 0b1010_0101_0000_1111;
    combinebits!(a, "aaaa aaaa", taco);
    combinebits!(a, "aaaa aaaa", "aaaa aaaa");
}
