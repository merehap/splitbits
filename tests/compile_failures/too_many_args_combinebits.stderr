error: proc macro panicked
 --> tests/compile_failures/too_many_args_combinebits.rs:5:5
  |
5 |     combinebits!(a, "aaaa aaaa", taco);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: The template expression must come after the input value(s), and must be a literal, but found:
          taco

error: proc macro panicked
 --> tests/compile_failures/too_many_args_combinebits.rs:6:5
  |
6 |     combinebits!(a, "aaaa aaaa", "aaaa aaaa");
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: Only one template must be present, but found this string literal too: 'aaaa aaaa'.
